code/code/sys/CharacterList.cc

```diff
@@ -0,0 +1,47 @@
+#include "CharacterList.h"
+#include "being.h"
+#include "connect.h"
+
+extern TBeing *character_list;
+
+TCharacterList::iterator::iterator(TBeing* d)
+  : d(d)
+{}
+
+TCharacterList::iterator TCharacterList::iterator::operator++()
+{
+  d = d->next;
```

Comment:
Won't we see a crash here in the case the being that d points to is deleted?

Reply:
We would, but modifying the loop range while looping is a really, really bad idea anyway. I hope we'll never do that again.